### PR TITLE
[zh] Recurrent Neural Networks

### PR DESCRIPTION
**Pull request by HsinJhao**
cs-230 recurrent neural networks translation is finished.

### CODE REVIEW

**Comment by shervinea** [comment]
Thank you for all your work @HsinJhao! Please feel free to invite anyone you know to review the translation.

File: .history/zh/cs-230-recurrent-neural-networks_20191006214850.md [hunk 1]
```diff
@@ -0,0 +1,676 @@
...
+
+&#10230;
+[优点, 可处理任何长度的输入, 模型大小不会随输入大小增加, 计算考虑历史信息, 权重在时间尺度上被整个网络共享]
+<br>
+
+
+**17. [Drawbacks, Computation being slow, Difficulty of accessing information from a long time ago, Cannot consider any future input for the current state]**
+
+&#10230;
+[缺点, 计算缓慢, 难以访问长时间的历史信息, 难以考虑未来时间步的输入对当前状态的影响]
```
Comment (SpeakingTom) [inline .history/zh/cs-230-recurrent-neural-networks_20191006214850.md]: 无法考虑未来时间步的输入对当前状态的影响

File: .history/zh/cs-230-recurrent-neural-networks_20191006214850.md [hunk 2]
```diff
@@ -0,0 +1,676 @@
...
+&#10230;
+
+<br>[波束搜索故障，RNN故障，增加波束宽度，尝试不同架构，正则化，获取更多数据] 
+
+
+**81. Bleu score ― The bilingual evaluation understudy (bleu) score quantifies how good a machine translation is by computing a similarity score based on n-gram precision. It is defined as follows:**
+
+&#10230;
+
+<br>bleu分数 ― 双语评估替补(bilingual evaluation understudy, bleu)分数通过基于n-gram精度计算相似度分数来量化机器翻译的好坏。其定义如下：
```
Comment (SpeakingTom) [inline .history/zh/cs-230-recurrent-neural-networks_20191006214850.md]: 双语评估替补？？？
通过基于n-gram精度计算相似性分数来量化机器翻译的质量。

File: .history/zh/cs-230-recurrent-neural-networks_20191006214850.md [hunk 3]
```diff
@@ -0,0 +1,676 @@
...
+&#10230;
+
+<br>[以ow表示, 朴素方法, 没有相似信息, 以ew表示, 考虑词汇之间的相似性]
+
+
+**48. Embedding matrix ― For a given word w, the embedding matrix E is a matrix that maps its 1-hot representation ow to its embedding ew as follows:**
+
+&#10230;
+
+<br>嵌入矩阵 - 对于给定的词汇w, 将该词汇的one-hot表示ow映射至词嵌入表示ew的嵌入矩阵E满足下式：
```
[the remark this answers is not in the document]
Comment (HsinJhao) [reply .history/zh/cs-230-recurrent-neural-networks_20191006214850.md]: Without changing the original intention, it will be translated to:
'' 嵌入矩阵 - 对于给定的词汇w, 通过嵌入矩阵E可将该词汇的one-hot表示向量ow映射为词嵌入表示向量ew, E满足下式：''

File: .history/zh/cs-230-recurrent-neural-networks_20191006214850.md (hunk 2 above)
Comment (HsinJhao) [reply .history/zh/cs-230-recurrent-neural-networks_20191006214850.md]: Through searching the related chinese information，**'bilingual evaluation understudy score'** will be translated to **双语评估替换分数**.

通过查阅相关中文资料，将**'bilingual evaluation understudy score'**翻译为**双语评估替换分数**为宜.

File: .history/zh/cs-230-recurrent-neural-networks_20191006214850.md [hunk 4]
```diff
@@ -0,0 +1,676 @@
...
+
+&#10230;
+梯度消失/梯度爆炸 - 梯度消失和梯度爆炸现象常出现在RNN模型中。其原因是该模型结构难以捕获长期依赖性, 因为乘法梯度会随着层数增加而呈指数递减/递增。
+<br>
+
+
+**28. Gradient clipping ― It is a technique used to cope with the exploding gradient problem sometimes encountered when performing backpropagation. By capping the maximum value for the gradient, this phenomenon is controlled in practice.**
+
+&#10230;
+梯度裁剪 - 该方法是用于解决进行反向传播时时而出现梯度爆炸问题的技术。通过限制梯度的最大值, 这种现象在实际中得到了相应的控制。
```
[the remark this answers is not in the document]
Comment (HsinJhao) [reply .history/zh/cs-230-recurrent-neural-networks_20191006214850.md]: To be consistent with the chinese version of book 《deep learning》(lan Goodfellow, et al),**Gradient clipping** will be translated to *梯度截断* rather than **梯度裁剪**.

为与Ian Goodfellow等人编写的《deep learning》中文译本保持一致，将**Gradient clipping**翻译为**梯度截断**而不是**梯度裁剪**。

File: .history/zh/cs-230-recurrent-neural-networks_20191006214850.md [hunk 5]
```diff
@@ -0,0 +1,676 @@
...
+
+&#10230;
+一个典型的RNN体系结构的优点和缺点可概括如下表：
+<br>
+
+
+**16. [Advantages, Possibility of processing input of any length, Model size not increasing with size of input, Computation takes into account historical information, Weights are shared across time]**
+
+&#10230;
+[优点, 可处理任何长度的输入, 模型大小不会随输入大小增加, 计算考虑历史信息, 权重在时间尺度上被整个网络共享]
```
[the remark this answers is not in the document]
Comment (HsinJhao) [reply .history/zh/cs-230-recurrent-neural-networks_20191006214850.md]: Applied, thanks!

File: .history/zh/cs-230-recurrent-neural-networks_20191006214850.md [hunk 6]
```diff
@@ -0,0 +1,676 @@
...
+
+&#10230;
+并且
+<br>
+
+
+**14. where Wax,Waa,Wya,ba,by are coefficients that are shared temporally and g1,g2 activation functions.**
+
+&#10230;
+其中Wax,Waa,Wya,ba是相关的系数矩阵, 在时间尺度上被整个网络共享；g1,g2是相关的激活函数。
```
[the remark this answers is not in the document]
Comment (HsinJhao) [reply .history/zh/cs-230-recurrent-neural-networks_20191006214850.md]: Applied!

**Comment by shervinea** [comment]
Thank you @HsinJhao and @SpeakingTom for all your work!

@HsinJhao: would it be possible to only keep the `zh/cs-230-recurrent-neural-networks.md` file in this PR? That way, we could immediately move forward with the merge.

**Comment by shervinea** [comment]
Thank you @HsinJhao and @SpeakingTom for all your work!

@HsinJhao: would it be possible to only keep the `zh/cs-230-recurrent-neural-networks.md` file in this PR? That way, we could immediately move forward with the merge.
